lib/html_to_plain_text.rb

```diff
@@ -66,19 +73,22 @@ def convert_node_to_plain_text(parent, out = "", options = {})
           out << node.text
         elsif node.element? && !IGNORE_TAGS.include?(node.name)
           convert_node_to_plain_text(node, out, child_options(node, options))
-          
+
           if node.name == BR
-            out.sub!(TRAILING_WHITESPACE, '')
-            out << "\n"
+            out.sub!(TRAILING_WHITESPACE, EMPTY)
+            out << NEWLINE
           elsif node.name == HR
-            out.sub!(TRAILING_WHITESPACE, '')
-            out << "\n" unless out.end_with?("\n")
+            out.sub!(TRAILING_WHITESPACE, EMPTY)
+            out << NEWLINE unless out.end_with?(NEWLINE)
             out << "-------------------------------\n"
           elsif node.name == TD || node.name == TH
             out << " | "
           elsif node.name == A
-            href = node["href"]
-            if href && href.match(ABSOLUTE_URL_PATTERN) && node.text.match(/\S/)
+            href = node[HREF]
+            if href &&
+                href =~ ABSOLUTE_URL_PATTERN &&
+                node.text =~ NOT_WHITESPACE_PATTERN &&
+                node.text != href[NON_PROTOCOL_PATTERN, 1]
```

Comment:
It's not obvious to me what this check is here for. Could you add a comment to the code explaining what the logic should be doing.